carmajava/launch/localization.launch

```diff
@@ -0,0 +1,84 @@
+  the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations under
+  the License.
+-->
+<!--
+	Launch file for launching the nodes in the CARMA localization stack minus drivers
+-->
+<launch>
+
+  <arg name="INTR_NS" default="hardware_interface" doc="Namespace of nodes hardware interface stack"/>
+
+  <!-- Remap any absolute topics to be relative -->
+  <remap from="/vehicle/odom" to="vehicle/odom"/>
+  <remap from="/estimate_twist" to="estimate_twist"/>
+  <remap from="/estimated_vel" to="estimated_vel"/>
+  <remap from="/estimated_vel_kmph" to="estimated_vel_kmph"/>
+  <remap from="/estimated_vel_mps" to="estimated_vel_mps"/>
+  <remap from="/filtered_points" to="filtered_points"/>
+  <remap from="/imu_raw" to="imu_raw"/>
+  <remap from="/localizer_pose" to="localizer_pose"/>
+  <remap from="/ndt_pose" to="ndt_pose"/>
+  <remap from="/ndt_reliability" to="ndt_reliability"/>
+  <remap from="/ndt_stat" to="ndt_stat"/>
+  <remap from="/pmap_stat" to="pmap_stat"/>
+  <remap from="/points_downsampler_info" to="points_downsampler_info"/>
+  <remap from="/points_map" to="points_map"/>
+  <remap from="/predict_pose" to="predict_pose"/>
+  <remap from="/predict_pose_imu" to="predict_pose_imu"/>
+  <remap from="/predict_pose_imu_odom" to="predict_pose_imu_odom"/>
+  <remap from="/predict_pose_odom" to="predict_pose_odom"/>
+  <remap from="/time_ndt_matching" to="time_ndt_matching"/>
+
+  <!-- Remap topics from external packages -->
+  <remap from="imu_raw" to="$(arg INTR_NS)/imu/imu_raw"/>
+
+  <remap from="system_alert" to="/system_alert"/>
+
+  <!-- <arg name="current_twist" default="current_twist" doc="Remapping of the twist topic used by the deadreckoner node"/>
+  <arg name="current_odom" default="current_odom" doc="Remapping of the twist topic used by the deadreckoner node"/> -->
+  <!-- Localization Package -->
+  <!-- Point Cloud Map TODO file location and parameter loading process -->
+  <node pkg="map_file" type="points_map_loader" name="points_map_loader" args="noupdate $(find carma)/map/base_map.pcd" />
+
+  <!-- NDT Matching-->
+  <!-- This namesapce sets the parameters which are not set by default in the ndt_matching.launch file -->
+  <!-- These parameters are not in the ndt_matching node private namespace-->
+  <!-- TODO resolve code duplication. These parameters already exist in base_link->velodyne tf -->
+
+  <param name="localizer" value="velodyne" />
+  <param name="tf_x" value="1.02" />
+  <param name="tf_y" value="0.0" />
+  <param name="tf_z" value="2.23" />
+  <param name="tf_roll" value="0.0" />
+  <param name="tf_pitch" value="0.0" />
+  <param name="tf_yaw" value="0.0" />
```

Comment:
Should we put those tf parameters in a urdf file?

Reply:
They are actually already in their under the Lidar tag, but the ndt_matching node doesn't use the tf library and reads them as parameters instead. Ideally I would like to change this, but I think that sort of modification should be included in a holistic approach to parameters from autoware.